pkg/clusterlabels/labels.go

```diff
@@ -0,0 +1,36 @@
+package clusterlabels
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"strings"
+
+	"github.com/giantswarm/kubectl-gs/internal/label"
+)
+
+// Logic partially lifted from https://github.com/kubernetes/kubectl/blob/445ad13/pkg/cmd/label/label.go#L400-L425
+// Licensed under Apache-2.0
+func Parse(rawLabels []string) (map[string]string, error) {
+	labels := map[string]string{}
+	for _, labelSpec := range rawLabels {
+		if strings.Contains(labelSpec, "=") {
+			parts := strings.Split(labelSpec, "=")
+			if len(parts) != 2 {
+				return nil, fmt.Errorf("invalid label spec: %v", labelSpec)
+			}
+			if strings.Contains(parts[0], label.ForbiddenLabelKeyPart) {
+				return nil, fmt.Errorf("invalid label key: %q: containing forbidden substring '%s'", labelSpec, label.ForbiddenLabelKeyPart)
+			}
+			if errs := validation.IsQualifiedName(parts[0]); len(errs) != 0 {
+				return nil, fmt.Errorf("invalid label key: %q: %s", labelSpec, strings.Join(errs, ";"))
+			}
+			if errs := validation.IsValidLabelValue(parts[1]); len(errs) != 0 {
+				return nil, fmt.Errorf("invalid label value: %q: %s", labelSpec, strings.Join(errs, ";"))
+			}
+			labels[parts[0]] = parts[1]
+		} else {
+			return nil, fmt.Errorf("unknown label spec: %v", labelSpec)
+		}
```

Comment:
Is checking if the label contains the `=` token really necessary? We already check if the label format works without it.

```suggestion		
		parts := strings.Split(labelSpec, "=")
		if len(parts) != 2 {
			return nil, fmt.Errorf("invalid label spec: %v", labelSpec)
		}
		if strings.Contains(parts[0], label.ForbiddenLabelKeyPart) {
			return nil, fmt.Errorf("invalid label key: %q: containing forbidden substring '%s'", labelSpec, label.ForbiddenLabelKeyPart)
		}
		if errs := validation.IsQualifiedName(parts[0]); len(errs) != 0 {
			return nil, fmt.Errorf("invalid label key: %q: %s", labelSpec, strings.Join(errs, ";"))
		}
		if errs := validation.IsValidLabelValue(parts[1]); len(errs) != 0 {
			return nil, fmt.Errorf("invalid label value: %q: %s", labelSpec, strings.Join(errs, ";"))
		}
		labels[parts[0]] = parts[1]
```

Reply:
We would lose stack traces here by using `fmt.Errorf`, instead of `microerror.Maskf`

Reply:
I used `microerror.Mask(fmt.Errorf(...))` now because the errors are created here.

Reply:
We usually define errors in a separate file, in a format such as:

```go
var invalidLabelKeyError = &microerror.Error{
	Kind: "invalidLabelKeyError",
}

func IsInvalidLabelKey(err error) bool {
	return microerror.Cause(err) == invalidLabelKeyError
}
```

Which you would then call as:

```go
microerror.Maskf(invalidLabelKeyError, %q: containing forbidden substring '%s'", labelSpec, label.ForbiddenLabelKeyPart)
```